network/netplay/netplay_frontend.c

```diff
@@ -8587,15 +8638,16 @@ static bool kick_client_by_id_and_name(netplay_t *netplay,
  */
 bool netplay_driver_ctl(enum rarch_netplay_ctl_state state, void *data)
 {
-   settings_t *settings       = config_get_ptr();
+   static bool guard = false;
```

Comment:
Question - instead of doing this internally in the function, can we instead pass this variable by pointer to the function, so that the callee controls how it's used instead? That would allow for more flexibility.

Reply:
I tried to remove the original guard, but found that you can't remove it without overhauling a good chunk of the external netplay code.

What you are asking isn't easily done either; here is an example.

1. netplay_driver_ctl is called externally with RARCH_NETPLAY_CTL_POST_FRAME.
2. netplay_post_frame is called by netplay_driver_ctl.
3. netplay_sync_post_frame is called by netplay_post_frame.
4. Under certain conditions, netplay_sync_post_frame may call core_unserialize.
5. core_unserialize will call netplay_driver_ctl with RARCH_NETPLAY_CTL_LOAD_SAVESTATE.
6. At this point, we do not want netplay_driver_ctl to load a savestate, because netplay_sync_post_frame is already handling it, and this is the sole purpose of this guard. Preventing recusive calls from doing anything.

Originally, the variable was in the net state struct, but I've made it a static storage duration variable to clean up that struct.

Reply:
Alright, we'll leave it at this then.